### PR TITLE
Fix for LocalKubernetesExecutor scheduler is not serving logs

### PR DESCRIPTION
**Pull request by snjypl**
```
*** Fetching from: http://airflow-scheduler-0.airflow-scheduler.airflow.svc.cluster.local:8793/log/dag_id=example_bash_operator/run_id=manual__2022-12-29T08:33:49.310825+00:00/task_id=runme_1/attempt=1.log
*** Failed to fetch log file from worker. [Errno 111] Connection refused
```

For `LocalKubernetesExecutor` we are not able to fetch the logs for local tasks. 

the scheduler is not starting the `serve_logs` process. 

https://github.com/apache/airflow/blob/94b3b897e2f94902777c4b24fb10c915279d8967/airflow/cli/commands/scheduler_command.py#L83-L87

we need to set `is_local` to `true` for `LocalKubernetesExecutor` 
https://github.com/apache/airflow/blob/94b3b897e2f94902777c4b24fb10c915279d8967/airflow/executors/local_kubernetes_executor.py#L46

also in the helm chart, when the executor is `LocalKubernetesExecutor` we need to create the `scheduler service` so that the tasks logs can be fetched. 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by dstandish** [comment]
How about we create flag `serve_logs` and add this to base executor and use _this_ instead of `is_local` in all cases?

CC @potiuk @o-nikolas

File: chart/templates/scheduler/scheduler-service.yaml
Comment (potiuk) [inline chart/templates/scheduler/scheduler-service.yaml]: I think if we want to make our Chart fully compatible with AIP-51 and allow "any" executors we should also implement some way to override Helm Chart behaviour for custom executors. But we can do it later. CC: @o-nikolas

Comment (potiuk) [reply chart/templates/scheduler/scheduler-service.yaml]: Ah.. I see @dstandish already had some comments on that one too :) Yeah - serve_logs should be fine and it should have default value depending on the "built-in" executors - so that we do not have to configure it manually for built-in executors.

**Comment by snjypl** [comment]
@potiuk  i noticed that `airflow-worker` service account is not getting created for `LocalKubernetesExecutor`.  

shall i create a new PR with all the helm chart related changes? or it can be part of this PR? 

```
kubernetes.client.exceptions.ApiException: (403)
Reason: Forbidden
HTTP response headers: HTTPHeaderDict({'Audit-Id': '6000d700-1b7b-413b-a49f-3eba7114bbf7', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Kubernetes-Pf-Flowschema-Uid': 'e6bfeec0-ef68-4420-88d8-b7024eed9fea', 'X-Kubernetes-Pf-Prioritylevel-Uid': '0e598fd5-a817-46f7-be85-6ac94aeb7872', 'Date': 'Thu, 29 Dec 2022 18:57:06 GMT', 'Content-Length': '400'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"example-local-kubernetes-executor-task-with-kubernetes-c5u03pq1\" is forbidden: error looking up service account airflow/airflow-worker: serviceaccount \"airflow-worker\" not found","reason":"Forbidden","details":{"name":"example-local-kubernetes-executor-task-with-kubernetes-c5u03pq1","kind":"pods"},"code":403}
```

**Comment by o-nikolas** [comment]
Hey folks!

I definitely agree with splitting out `is_local` into more fine grained compatibility flags/properties ([see here](https://github.com/apache/airflow/issues/28276#issuecomment-1371553409) for more context). Also expect an AIP soon which will deliver native support for hybrid executors (of any combination) rather than these handbuilt classes which combine two explicit executors :rocket:

**Comment by snjypl** [comment]
@dstandish  i have added `serve_logs` flag

File: airflow/executors/local_kubernetes_executor.py
Comment (o-nikolas) [inline airflow/executors/local_kubernetes_executor.py:51]: You'll need to add a default for the `CeleryKubernetesExecutor` as well (with False being the value I reckon), since that class does not inherit (directly) from BaseExecutor (see [this comment](https://github.com/apache/airflow/issues/28276#issuecomment-1344899475) for more details).

**Review by o-nikolas — CHANGES_REQUESTED** [review]

File: airflow/executors/local_kubernetes_executor.py
Comment (dstandish) [reply airflow/executors/local_kubernetes_executor.py:51]: i think all Celery* executors need to serve logs, so i'd assume (without having looked) that this one needs to be true (unless something weird happens and  it ends up serving anyway because of its child executor)

Comment (dstandish) [reply airflow/executors/local_kubernetes_executor.py:51]: i think maybe only k8s executor doesn't log serve... we read the pod logs instead (while task running)

Comment (snjypl) [reply airflow/executors/local_kubernetes_executor.py:51]: for `celery executor` the logs are served by the `workers`. 
https://github.com/apache/airflow/blob/3ececb2c79307bd943aad116d7b0b5933af8185a/airflow/cli/commands/celery_command.py#L88-L96

this flag check if the logs need to be served by the `scheduler`.   
should we rename this flag to `scheduler_serve_logs`?

Comment (snjypl) [reply airflow/executors/local_kubernetes_executor.py:51]: > You'll need to add a default for the `CeleryKubernetesExecutor` as well (with False being the value I reckon), since that class does not inherit (directly) from BaseExecutor (see [this comment](https://github.com/apache/airflow/issues/28276#issuecomment-1344899475) for more details).

should we do `hasattr(executor.serve_logs)` to be sure?

**Comment by snjypl** [comment]
> @potiuk i noticed that `airflow-worker` service account is not getting created for `LocalKubernetesExecutor`.
> 
> shall i create a new PR with all the helm chart related changes? or it can be part of this PR?
> 
> ```
> kubernetes.client.exceptions.ApiException: (403)
> Reason: Forbidden
> HTTP response headers: HTTPHeaderDict({'Audit-Id': '6000d700-1b7b-413b-a49f-3eba7114bbf7', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Kubernetes-Pf-Flowschema-Uid': 'e6bfeec0-ef68-4420-88d8-b7024eed9fea', 'X-Kubernetes-Pf-Prioritylevel-Uid': '0e598fd5-a817-46f7-be85-6ac94aeb7872', 'Date': 'Thu, 29 Dec 2022 18:57:06 GMT', 'Content-Length': '400'})
> HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods \"example-local-kubernetes-executor-task-with-kubernetes-c5u03pq1\" is forbidden: error looking up service account airflow/airflow-worker: serviceaccount \"airflow-worker\" not found","reason":"Forbidden","details":{"name":"example-local-kubernetes-executor-task-with-kubernetes-c5u03pq1","kind":"pods"},"code":403}
> [2022-12-29T18:57:06.871+0000] {kubernetes_executor.py:686} WARNING - ApiException when attempting to run task, re-queueing. Reason: 'Forbidden'. Message: pods "example-local-kubernetes-executor-task-with-kubernetes-c5u03pq1" is forbidden: error looking up service account airflow/airflow-worker: serviceaccount "airflow-worker" not found
> ```
@Chen-Oliver has opened a PR for this particular issue: https://github.com/apache/airflow/pull/28813

File: airflow/executors/local_kubernetes_executor.py
Comment (dstandish) [reply airflow/executors/local_kubernetes_executor.py:51]: Oh, durr

Comment (o-nikolas) [reply airflow/executors/local_kubernetes_executor.py:51]: > > You'll need to add a default for the `CeleryKubernetesExecutor` as well (with False being the value I reckon), since that class does not inherit (directly) from BaseExecutor (see [this comment](https://github.com/apache/airflow/issues/28276#issuecomment-1344899475) for more details).
> 
> should we do `hasattr(executor.serve_logs)` to be sure?


@snjypl 
I'd prefer to have a default added to the CeleryKubernetesExecutor so that code authors in the future don't need to remember/know that access to that flag needs to be protected with hasattr, but if you'd like to do both, then I'm fine with that as well!

Comment (snjypl) [reply airflow/executors/local_kubernetes_executor.py:51]: @o-nikolas  i have made the changes.

**Review by o-nikolas — APPROVED** [review]

**Comment by o-nikolas** [comment]
Anyone have some time to give this a second review/approval? Would be nice to get this merged for @snjypl
Maybe @potiuk, @dstandish or @pierrejeambrun?

**Review by pierrejeambrun — APPROVED** [review]
LGTM

**Comment by snjypl** [comment]
> Could use a rebase before merging, it's 20+ commits behind master

Done.

**Pull request merged**
